### PR TITLE
Change setLocalDescription to require unchanged offer/answer string

### PR DESCRIPTION
**Pull request by aboba**
Fix for Issue https://github.com/w3c/webrtc-pc/issues/950

### CODE REVIEW

File: webrtc.html [hunk 1]
```diff
@@ -1639,10 +1639,42 @@
...
+                  does not exactly match <var>lastOffer</var>,
+                  reject the promise with <code>InvalidParameters</code>
+                  and abort these steps.</li>
+                  <li>If <code><var>description</var>.type</code> is
+                  <code>answer</code> or <code>pranswer</code> and
+                  <code><var>description</var>.sdp</code> does not exactly
+                  match <var>lastAnswer</var>, reject the promise with
+                  <code>InvalidParameters</code> and abort these steps.</li>               
+                  <li>Return the result of <a href="#set-description">setting
+                  the RTCSessionDescription</a> indicated by the method's first
```
Comment (taylor-b) [inline webrtc.html]: Should say either "the argument" or "the first argument" consistently

File: webrtc.html [hunk 2]
```diff
@@ -1639,10 +1639,42 @@
...
+                  is <code>answer</code> or <code>pranswer</code>,
+                  set <code><var>description</var>.sdp</code> to
+                  <var>lastAnswer</var>.</li>
+                  <li>If <code><var>description</var>.sdp</code> is
+                  <code>null</code> and <code><var>description</var>.type</code>
+                  is <code>offer</code>, set <code><var>description</var>.sdp</code>
+                  to <var>lastOffer</var>.</li>
+                  <li>If <code><var>description</var>.type</code> is
+                  <code>offer</code> and <code><var>description</var>.sdp</code>
+                  does not exactly match <var>lastOffer</var>,
```
Comment (taylor-b) [inline webrtc.html]: What went into JSEP isn't that the SDP must be *exactly* the same, but that the parsed representation must be the same. For example, "a=" lines appearing in a different order shouldn't matter.

Maybe we could simplify this by just referring to JSEP:

> If `description.sdp` is not `null`, and the browser determines that it was modified after the last call to `createOffer`/`createAnswer`, as described by JSEP Section 5.8, reject `p` with an `InvalidAccessError` and abort these steps.

File: webrtc.html [hunk 3]
```diff
@@ -1639,10 +1639,42 @@
...
+                  reject the promise with <code>InvalidParameters</code>
```
Comment (taylor-b) [inline webrtc.html]: We use `InvalidAccessError` for invalid parameters. Or maybe `InvalidModificationError` would be more appropriate? I'm not sure.

File: webrtc.html (hunk 2 above)
Comment (aboba) [inline webrtc.html]: I posted a question to the RTCWEB mailing list on this.

File: webrtc.html (hunk 3 above)
Comment (aboba) [inline webrtc.html]: Elsewhere, InvalidModificationError is used, so I changed it to that.

File: webrtc.html (hunk 1 above)
Comment (aboba) [inline webrtc.html]: Using "first argument" throughout.

**Pull request merged**